tests/installation/module_registration/register_module_regpage.pm

```diff
@@ -0,0 +1,41 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: After register Application module
+#          in "Extension and Module Selection" dialog
+#          has register page
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use registration;
+use testapi qw(save_screenshot get_var get_required_var);
+
+our %ADDONS_REGCODE = (
+    'sle-ha' => get_var('SCC_REGCODE_HA'),
+    'sle-ha-geo' => get_var('SCC_REGCODE_GEO'),
+    'sle-we' => get_var('SCC_REGCODE_WE'),
+    'sle-module-live-patching' => get_var('SCC_REGCODE_LIVE'),
+    'sle-live-patching' => get_var('SCC_REGCODE_LIVE'),
+    'SLES-LTSS' => get_var('SCC_REGCODE_LTSS'),
+    'SUSE-Linux-Enterprise-RT' => get_var('SCC_REGCODE_RT'),
+    ESPOS => get_var('SCC_REGCODE_ESPOS'),
+);
+
+sub run {
+    my @scc_addons = grep($_, split(/,/, get_required_var('SCC_ADDONS')));
```

Comment:
you just need after this line a `grep /we|ha| etc./ @scc_addons` which will give you a list empty if the job doesn't have it or a list with the elements that requires additional regcode. Then you don't need to get the full name at all, your hash can have as keys the addons keys, `we, ha, etc.`